src/term/transform/encode_strs.rs

```diff
@@ -80,3 +81,19 @@ impl Term {
     }
   }
 }
+
+impl Pattern {
+  pub fn encode_str(&mut self) -> bool {
+    match self {
+      Pattern::Var(Some(Name(nam))) => nam == SCONS || nam == SNIL,
+      Pattern::Ctr(Name(nam), pats) => {
+        let mut uses = nam == SCONS || nam == SNIL;
+        for pat in pats {
+          uses |= pat.encode_str();
+        }
+        uses
+      }
+      _ => false,
```

Comment:
missing Pattern::Tup & Pattern::List here.

also, can use this function instead of this line on the two files
```rs
    pat.names().chain(pat.ctrs()).any(|Name(n)| matches!(n.as_str(), ...));
```